src/ui/SimpleFilter/SimpleFilter.ts

```diff
@@ -0,0 +1,188 @@
+        this.element.appendChild(multiselect.el);
+
+        this.checkboxContainer = $$('div', { className: 'checkboxes'});
+        this.checkboxes = _.map(this.options.values, (value)=> this.createCheckboxes(value));
+        _.each(this.checkboxes, (result) => {
+            this.checkboxContainer.el.appendChild(result.checkbox.getElement())
+        });
+        this.element.appendChild(this.checkboxContainer.el);
+
+        $$(document.body).on('click', (e: Event)=> {
+            let target = <Element>e.target;
+
+            if( overselect.el == e.target){
+                this.onClick()
+            } else {
+                while (target != this.element) {
+                    if(target != this.root && target.parentElement != null) {
+                        target = target.parentElement;
+                    }
+                    if ((target == this.root || target.parentElement == null) && this.expanded) {
+                        this.onClick();
+                        break;
+                    } else if ((target == this.root || target.parentElement == null) && !this.expanded){
+                        break;
+                    }
+                }
+            }
+        });
+        this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+        this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.groupBy(args));
+        this.bind.onRootElement(QueryEvents.doneBuildingQuery, (args: IDoneBuildingQueryEventArgs) => this.handleDoneBuildingQuery(args));
+
+    }
+
+    private handleBuildingQuery(args: IBuildingQueryEventArgs) {
+        Assert.exists(args);
+        Assert.exists(args.queryBuilder);
+        let selectedValues = this.getSelectedValues();
+        if(selectedValues.length > 0) {
+            args.queryBuilder.advancedExpression.addFieldExpression(this.options.field.toLocaleString(), '==', selectedValues);
+        }
+    }
+
+    public onClick () {
+
+        if(!this.expanded){
+            this.checkboxContainer.el.style.display = 'block';
+            this.expanded = true;
+        } else {
+            this.checkboxContainer.el.style.display = 'none';
+            this.expanded = false;
+        }
+    }
+
+    private onCheck() {
+        this.queryController.executeQuery();
+    }
+
+    private createCheckboxes (value: string) {
+        const checkbox = new Checkbox(() => {
+            this.onCheck();
+        }, l(this.getValueCaption(value)));
+        return {checkbox, value};
+    }
+
+    public getValueCaption(value: string): string {
+        let lookupValue = value;
+        let ret = lookupValue;
+
+        if (Utils.exists(this.options.valueCaption)) {
+            if (typeof this.options.valueCaption == 'object') {
+                ret = this.options.valueCaption[lookupValue] || ret;
+            }
+        }
+        return ret;
+    }
+
+    public getSelected() {
```

Comment:
It's a good practice to explicity declare the return type for a function, especially if it's using underscore.js stuff.

Much easier to read.